products/azure_rm_batchaccount/api.yaml

```
@@ -0,0 +1,170 @@
+--- !ruby/object:Api::Product
+name: Azure BatchAccount Management
+prefix: azure_rm_batchaccount
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: NotUsedInAzure
+scopes:
+  - NotUsedInAzure
+azure_namespace: 'Microsoft.Batch'
+azure_version: '2018-12-01'
+objects:
+  - !ruby/object:Api::Resource
+    name: BatchAccount
+    api_name: Batch
+    base_url: NotUsedInAzure
+
+    azure_sdk_definition: !ruby/object:Api::Azure::SDKDefinition
+      provider_name: batch
+      go_client_namespace: batch
+      go_client: batchAccountClient
+      python_client_namespace: azure.mgmt.batch
+      python_client: BatchManagementClient.batch_account
+      create: !ruby/object:Api::Azure::SDKOperationDefinition
+        async: true
+        go_func_name: Create
+        python_func_name: create
+        request:
+          'resourceGroupName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: resourceGroups
+            python_variable_name: resource_group_name
+            go_variable_name: resourceGroup
+          'accountName': !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            id_portion: batchAccounts
+            python_variable_name: account_name
+            go_variable_name: accountName
+          '/': !ruby/object:Api::Azure::SDKTypeDefinition::ComplexObject
+            go_type_name: Properties
+            go_variable_name: parameters
+          /location: !ruby/object:Api::Azure::SDKTypeDefinition::StringObject
+            go_field_name: location
+          /tags: !ruby/object:Api::Azure::SDKTypeDefinition::ComplexObject
+            go_field_name: tags
+          /properties/autoStorage: !ruby/object:Api::Azure::SDKTypeDefinition::ComplexObject
+            go_field_name: autoStorage

```

For complex types, we also need to know the `go_type_name` defined by Go SDK as well.

```suggestion
            go_field_name: AutoStorage
            go_type_name: AutoStorageBaseProperties
```